### PR TITLE
Dedupe impressions into development

### PR DESCRIPTION
**Pull request by sanzmauro**
# .NET SDK

## Extra Notes

### CODE REVIEW

**Review by NicoZelaya — APPROVED** [review]

File: src/Splitio-net-core/Domain/ConfigurationOptions.cs [hunk 1]
```diff
@@ -33,5 +34,6 @@ public class ConfigurationOptions
         public int? StreamingReconnectBackoffBase { get; set; }
         public string AuthServiceURL { get; set; }
         public string StreamingServiceURL { get; set; }
+        public ImpressionMode? ImpressionMode { get; set; }
```
Comment (mmelograno) [inline src/Splitio-net-core/Domain/ConfigurationOptions.cs:37]: according to the spec it is `ImpressionsMode`

File: src/Splitio-net-core/Services/Cache/Lru/CacheEnumerator.cs [hunk 2]
```diff
@@ -0,0 +1,140 @@
+﻿// Copyright (c) Mark Davis. All rights reserved.
```
Comment (mmelograno) [inline src/Splitio-net-core/Services/Cache/Lru/CacheEnumerator.cs:1]: I think that you are missing here: license and repo url

Comment (mmelograno) [reply src/Splitio-net-core/Services/Cache/Lru/CacheEnumerator.cs:1]: You can follow this as an example https://github.com/splitio/javascript-client/blob/development/src/utils/lrucache/linkedlist.js#L2

File: src/Splitio-net-core/Services/Cache/Lru/Node.cs [hunk 3]
```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Mark Davis. All rights reserved.
```
Comment (mmelograno) [inline src/Splitio-net-core/Services/Cache/Lru/Node.cs:1]: same here

File: src/Splitio-net-core/Services/Impressions/Classes/ImpressionsManager.cs [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+            _addPreviousTime = addPreviousTime;
+            _optimized = impressionMode == ImpressionMode.Optimized && addPreviousTime;
+            _impressionsObserver = impressionsObserver;
+        }
+
+        public KeyImpression BuildImpression(string matchingKey, string feature, string treatment, long time, long? changeNumber, string label, string bucketingKey)
+        {
+            var impression = new KeyImpression(matchingKey, feature, treatment, time, changeNumber, label, bucketingKey);
+
+            if (_addPreviousTime && _impressionsObserver != null)
```
Comment (mmelograno) [inline src/Splitio-net-core/Services/Impressions/Classes/ImpressionsManager.cs:37]: you could add this validation in the constructor and only evaluate one of this here (false or true)

File: src/Splitio-net-core/Services/Shared/Classes/ConfigService.cs [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+            selfRefreshingConfig.TreatmentLogRefreshRate = GetImpressionRefreshRate(selfRefreshingConfig.ImpressionMode, config.ImpressionsRefreshRate);
+
+            return selfRefreshingConfig;
+        }
+
+        private int GetMinimunAllowed(int value, int minAllowed, string configName)
+        {
+            if (value < minAllowed)
+            {
+                _log.Warn($"{configName} minumum allowed value: {minAllowed}");
```
Comment (mmelograno) [inline src/Splitio-net-core/Services/Shared/Classes/ConfigService.cs:89]: wording: minimum

**Review by mmelograno — CHANGES_REQUESTED** [review]

**Comment by sanzmauro** [comment]
@mmelograno changes applied!

**Review by mmelograno — APPROVED** [review]
LGTM!

**Pull request merged**